validator/client/validator.go

```diff
@@ -745,10 +752,12 @@ func (v *validator) RolesAt(ctx context.Context, slot primitives.Slot) (map[[fie
 		// the validator checks whether it's in the sync committee of following epoch.
 		inSyncCommittee := false
 		if slots.IsEpochEnd(slot) {
+			v.dutiesLock.RLock()
 			if v.duties.NextEpochDuties[validator].IsSyncCommittee {
 				roles = append(roles, iface.RoleSyncCommittee)
 				inSyncCommittee = true
 			}
+			v.dutiesLock.RUnlock()
```

Comment:
You already have a RLock at the top of this method.

This is probably what is causing the static analysis check. 

```
validator/client/validator.go:755:4: found recursive read lock call (recursivelock)
```

Reply:
@prestonvanloon thanks for the feedback. I've updated the locks and hope that there are no more recursive calls.